run-llama/llama_index
# Add option to include pipeline_id at top level in MongoDBAtlasVectorSearch

## Description
This PR introduces a new feature in the `MongoDBAtlasVectorSearch` class that allows users to add the `pipeline_id` as a top-level field in MongoDB documents, in addition to its existing presence within the metadata.

## Changes
- Added a new parameter `add_pipeline_id_top_level` (boolean) to the `MongoDBAtlasVectorSearch` constructor.
- Added a new parameter `pipeline_id_key` (string) to allow customization of the field name for the top-level pipeline ID.
- Modified the `add` method to include the pipeline_id at the top level of the document when the feature is enabled.

## Motivation
`pipeline_id` is required at top level in document, during retrieval from `llamacloud`, the `pipeline_id` is added as filter.

## Usage
Users can enable this feature as follows:

```python
vector_store = MongoDBAtlasVectorSearch(
    mongodb_client=client,
    add_pipeline_id_top_level=True,
    pipeline_id_key="custom_pipeline_id",  # Optional, defaults to "pipeline_id"
    # ... other parameters ...
)
```

## Testing
To test this change:
1. Initialize `MongoDBAtlasVectorSearch` with `add_pipeline_id_top_level=True`.
2. Add nodes with metadata containing a `pipeline_id`.
3. Verify that the inserted documents in MongoDB contain the `pipeline_id` at the top level.
4. Test with `add_pipeline_id_top_level=False` to ensure backward compatibility.
